test/model/typescript/duplicate-cat.ts

```diff
@@ -0,0 +1,20 @@
+import { Animal } from "./animal";
+import { JsonObject, JsonProperty } from "../../../src/json2typescript/json-convert-decorators";
+import { DateConverter } from "./date-converter";
+
+/**
+ * Class which has a property with the same name as Cat.  Used to test handling
+ * of prototype-based property mapping search.  Property has the same name, but
+ * an incompatible type, and is initialized (so it's present) but purposely left
```

Comment:
> Property has the same name, but an incompatible type

That's I exactly how I experienced the problem: I had two sibling classes that properties of the same name, but in one case the property was optional and in the other one it was required.

If the properties' types are compatible, the behavior is still wrong (definition is possibly taken from the wrong class), but it doesn't have any effect.

Reply:
So I think your test ensures that the logic works correctly now.